firmware/fw_base.S

```diff
@@ -175,6 +175,13 @@ _wait_for_boot_hart:
 	REG_L	a5, (a4)
 	beqz	a5, _wait_for_boot_hart
 
+	li	ra, 0
+	call	_reset_regs
+
+	csrr	tp, CSR_MSCRATCH
+	/* Setup stack */
+	add	sp, tp,zero
+
 _start_warm:
 	/* Disable and clear all interrupts */
```

Comment:
I think you missed my last comment.

We should call _reset_regs() in _start_warm() so that its done for all HART. This will also work for HART hotplug.

Reply:
argh. Yes. _reset_regs call was supposed to be after _start_warm. I think I messed up while switching to latest branch. Sorry.